core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -438,17 +376,15 @@ func (i *pluginOracleCreator) createReadersAndWriters(
 			return nil, nil, fmt.Errorf("failed to get chain selector from chain ID %s: %w", chainID, err1)
 		}
 
-		if _, exists := plugins[relayChainFamily]; !exists {
-			i.lggr.Debugw("createReadersAndWriters: skipping unsupported relayer", "chainID", chainID, "family", relayChainFamily)
-			continue
-		}
-
-		chainReaderConfig, err1 := getChainReaderConfig(i.lggr, chainID, destChainID, homeChainID, ofc, chainSelector, relayChainFamily)
-		if err1 != nil {
-			return nil, nil, fmt.Errorf("failed to get chain reader config: %w", err1)
-		}
-
-		cr, err1 := relayer.NewContractReader(ctx, chainReaderConfig)
+		cr, err1 := pluginConfig.GetChainReaderWriter.GetChainReader(ctx, ccipcommon.GetChainReaderParams{
```

Comment:
So looks like here, the CR will be created based on either chainID, or the destChainId. However the chainSelector points to the chain chainID, not the destChainId.

But look at line 425 now which says:
contractReaders[chainSelector] = cr

This is clearly wrong, in case the cr was created from destChainId.

This was a pre-existing error in the code, not in your PR. Maybe we can fix it first in a separate PR, and pollute this PR?